### ISSUE TITLE
Unable to run on a mill based project

### ISSUE THREAD

**Issue by AlexITC** [stryker-mutator/stryker4s#81]
### Summary

Trying to use a [project](https://github.com/AlexITC/playsonify) built with mill is throwing errors.

```
➜  stryker4s git:(master) ✗ sbt run
[info] Loading settings from idea.sbt,gpg.sbt ...
[info] Loading global plugins from /home/dell/.sbt/1.0/plugins
[info] Loading settings from plugins.sbt ...
[info] Loading project definition from /home/dell/projects/github/stryker4s/project
[info] Loading settings from build.sbt ...
[info] Set current project to stryker4s (in build file:/home/dell/projects/github/stryker4s/)
[info] Running stryker4s.run.Stryker4sRunner 
11:32:28.322 INFO  ConfigReader$: Setting logging level to INFO.
11:32:28.324 INFO  ConfigReader$: Using stryker4s.conf in the current working directory
11:32:28.420 INFO  Mutator: Found 0 of 0 file(s) to be mutated.
11:32:28.420 INFO  Mutator: 0 Mutant(s) generated
[error] (run-main-0) java.nio.file.FileSystemException: /home/dell/projects/mine/playsonify/out/mill-worker-1/io: No such device or address
[error] java.nio.file.FileSystemException: /home/dell/projects/mine/playsonify/out/mill-worker-1/io: No such device or address
[error] 	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:91)
[error] 	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
[error] 	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
[error] 	at sun.nio.fs.UnixCopyFile.copyFile(UnixCopyFile.java:230)
[error] 	at sun.nio.fs.UnixCopyFile.copy(UnixCopyFile.java:581)
[error] 	at sun.nio.fs.UnixFileSystemProvider.copy(UnixFileSystemProvider.java:253)
[error] 	at java.nio.file.Files.copy(Files.java:1274)
[error] 	at better.files.File$$anon$1.visitFile(File.scala:1020)
[error] 	at better.files.File$$anon$1.visitFile(File.scala:1011)
[error] 	at java.nio.file.Files.walkFileTree(Files.java:2670)
[error] 	at java.nio.file.Files.walkFileTree(Files.java:2742)
[error] 	at better.files.File.copyTo(File.scala:1011)
[error] 	at stryker4s.run.ProcessMutantRunner.apply(ProcessMutantRunner.scala:27)
[error] 	at stryker4s.Stryker4s.run(Stryker4s.scala:17)
[error] 	at stryker4s.run.Stryker4sRunner$.delayedEndpoint$stryker4s$run$Stryker4sRunner$1(Stryker4sRunner.scala:23)
[error] 	at stryker4s.run.Stryker4sRunner$delayedInit$body.apply(Stryker4sRunner.scala:12)
[error] 	at scala.Function0.apply$mcV$sp(Function0.scala:34)
[error] 	at scala.Function0.apply$mcV$sp$(Function0.scala:34)
[error] 	at scala.runtime.AbstractFunction0.apply$mcV$sp(AbstractFunction0.scala:12)
[error] 	at scala.App.$anonfun$main$1$adapted(App.scala:76)
[error] 	at scala.collection.immutable.List.foreach(List.scala:389)
[error] 	at scala.App.main(App.scala:76)
[error] 	at scala.App.main$(App.scala:74)
[error] 	at stryker4s.run.Stryker4sRunner$.main(Stryker4sRunner.scala:12)
[error] 	at stryker4s.run.Stryker4sRunner.main(Stryker4sRunner.scala)
[error] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[error] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[error] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[error] 	at java.lang.reflect.Method.invoke(Method.java:498)
[error] 	at sbt.Run.invokeMain(Run.scala:93)
[error] 	at sbt.Run.run0(Run.scala:87)
[error] 	at sbt.Run.execute$1(Run.scala:65)
[error] 	at sbt.Run.$anonfun$run$4(Run.scala:77)
[error] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
[error] 	at sbt.util.InterfaceUtil$$anon$1.get(InterfaceUtil.scala:10)
[error] 	at sbt.TrapExit$App.run(TrapExit.scala:252)
[error] 	at java.lang.Thread.run(Thread.java:748)
[error] java.lang.RuntimeException: Nonzero exit code: 1
[error] 	at sbt.Run$.executeTrapExit(Run.scala:124)
[error] 	at sbt.Run.run(Run.scala:77)
[error] 	at sbt.Defaults$.$anonfun$bgRunTask$5(Defaults.scala:1185)
[error] 	at sbt.Defaults$.$anonfun$bgRunTask$5$adapted(Defaults.scala:1180)
[error] 	at sbt.internal.BackgroundThreadPool.$anonfun$run$1(DefaultBackgroundJobService.scala:366)
[error] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
[error] 	at scala.util.Try$.apply(Try.scala:209)
[error] 	at sbt.internal.BackgroundThreadPool$BackgroundRunnable.run(DefaultBackgroundJobService.scala:289)
[error] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error] 	at java.lang.Thread.run(Thread.java:748)
[error] (Compile / run) Nonzero exit code: 1
[error] Total time: 1 s, completed Oct 6, 2018 11:32:28 AM

```
### Stryker config

```hocon
stryker4s {
  files: [
    "playsonify/src/**/*.scala"
  ]
   base-dir: "/home/dell/projects/mine/playsonify"
   log-level: INFO
   test-runner: {
    type: commandrunner
    command: "mill"
    args: "playsonify.test"
  }
   reporters: ["console"]
}
```

### Stryker4s environment

```
48dec70c9ddffcffaf8afde9b386d806188578c4
```

### Your Environment
| software             | version(s) |
| -------------------- | ---------- |
| Scala version        |       2.12     |
| Build tool & version |  mill 0.2.8          | 
| Operating System     |   ubuntu 16.04      |

**Comment by hugo-vrijswijk** [comment]
Could you try changing the `files` option to `mutate`? It looks like our documentation is a little out of date. I'll update it soon

**Comment by AlexITC** [comment]
Now I get this:
```
[error] (run-main-0) java.lang.UnsupportedOperationException: Invalid transformation of Case.pat: Lit.String -> Term.Match. From: Lit.String("asc"), to: Term.Match(Term.Apply(Term.Select(Term.Select(Term.Name("sys"), Term.Name("env")), Term.Name("get")), List(Lit.String("ACTIVE_MUTATION"))), List(Case(Pat.Extract(Term.Name("Some"), List(Lit.String("15"))), None, Lit.String("")), Case(Pat.Wildcard(), None, Lit.String("asc"))))
[error] java.lang.UnsupportedOperationException: Invalid transformation of Case.pat: Lit.String -> Term.Match. From: Lit.String("asc"), to: Term.Match(Term.Apply(Term.Select(Term.Select(Term.Name("sys"), Term.Name("env")), Term.Name("get")), List(Lit.String("ACTIVE_MUTATION"))), List(Case(Pat.Extract(Term.Name("Some"), List(Lit.String("15"))), None, Lit.String("")), Case(Pat.Wildcard(), None, Lit.String("asc"))))
[error] 	at scala.meta.transversers.Transformer.fail(Transformer.scala:4)
[error] 	at scala.meta.transversers.Transformer.apply(Transformer.scala:4)
[error] 	at stryker4s.extensions.TreeExtensions$ImplicitTreeExtensions$OnceTransformer.super$apply(TreeExtensions.scala:66)
[error] 	at stryker4s.extensions.TreeExtensions$ImplicitTreeExtensions$OnceTransformer.$anonfun$apply$1(TreeExtensions.scala:66)
[error] 	at scala.Option.getOrElse(Option.scala:121)
[error] 	at stryker4s.extensions.TreeExtensions$ImplicitTreeExtensions$OnceTransformer.apply(TreeExtensions.scala:66)
[error] 	at scala.meta.transversers.Transformer.apply(Transformer.scala:4)
[error] 	at stryker4s.extensions.TreeExtensions$ImplicitTreeExtensions$OnceTransformer.super$apply(TreeExtensions.scala:66)
[error] 	at stryker4s.extensions.TreeExtensions$ImplicitTreeExtensions$OnceTransformer.$anonfun$apply$1(TreeExtensions.scala:66)
[error] 	at scala.Option.getOrElse(Option.scala:121)
[error] 	at stryker4s.extensions.TreeExtensions$ImplicitTreeExtensions$OnceTransformer.apply(TreeExtensions.scala:66)
[error] 	at scala.meta.transversers.Transformer.apply(Transformer.scala:4)
[error] 	at stryker4s.extensions.TreeExtensions$ImplicitTreeExtensions$OnceTransformer.super$apply(TreeExtensions.scala:66)
[error] 	at stryker4s.extensions.TreeExtensions$ImplicitTreeExtensions$OnceTransformer.$anonfun$apply$1(TreeExtensions.scala:66)
[error] 	at scala.Option.getOrElse(Option.scala:121)
[error] 	at stryker4s.extensions.TreeExtensions$ImplicitTreeExtensions$OnceTransformer.apply(TreeExtensions.scala:66)
[error] 	at scala.meta.transversers.Transformer.apply(Transformer.scala:4)
[error] 	at stryker4s.extensions.TreeExtensions$ImplicitTreeExtensions$OnceTransformer.super$apply(TreeExtensions.scala:66)
[error] 	at stryker4s.extensions.TreeExtensions$ImplicitTreeExtensions$OnceTransformer.$anonfun$apply$1(TreeExtensions.scala:66)
[error] 	at scala.Option.getOrElse(Option.scala:121)
[error] 	at stryker4s.extensions.TreeExtensions$ImplicitTreeExtensions$OnceTransformer.apply(TreeExtensions.scala:66)
[error] 	at scala.meta.transversers.Transformer.apply(Transformer.scala:4)
[error] 	at stryker4s.extensions.TreeExtensions$ImplicitTreeExtensions$OnceTransformer.super$apply(TreeExtensions.scala:66)
[error] 	at stryker4s.extensions.TreeExtensions$ImplicitTreeExtensions$OnceTransformer.$anonfun$apply$1(TreeExtensions.scala:66)
[error] 	at scala.Option.getOrElse(Option.scala:121)
[error] 	at stryker4s.extensions.TreeExtensions$ImplicitTreeExtensions$OnceTransformer.apply(TreeExtensions.scala:66)
[error] 	at scala.meta.transversers.Transformer.apply(Transformer.scala:4)
[error] 	at stryker4s.extensions.TreeExtensions$ImplicitTreeExtensions$OnceTransformer.super$apply(TreeExtensions.scala:66)
[error] 	at stryker4s.extensions.TreeExtensions$ImplicitTreeExtensions$OnceTransformer.$anonfun$apply$1(TreeExtensions.scala:66)
[error] 	at scala.Option.getOrElse(Option.scala:121)
[error] 	at stryker4s.extensions.TreeExtensions$ImplicitTreeExtensions$OnceTransformer.apply(TreeExtensions.scala:66)
[error] 	at scala.meta.transversers.Transformer.apply(Transformer.scala:4)
[error] 	at stryker4s.extensions.TreeExtensions$ImplicitTreeExtensions$OnceTransformer.super$apply(TreeExtensions.scala:66)
[error] 	at stryker4s.extensions.TreeExtensions$ImplicitTreeExtensions$OnceTransformer.$anonfun$apply$1(TreeExtensions.scala:66)
[error] 	at scala.Option.getOrElse(Option.scala:121)
[error] 	at stryker4s.extensions.TreeExtensions$ImplicitTreeExtensions$OnceTransformer.apply(TreeExtensions.scala:66)
[error] 	at stryker4s.extensions.TreeExtensions$ImplicitTreeExtensions.transformOnce(TreeExtensions.scala:61)
[error] 	at stryker4s.mutants.applymutants.MatchBuilder.$anonfun$buildNewSource$1(MatchBuilder.scala:20)
[error] 	at scala.collection.LinearSeqOptimized.foldLeft(LinearSeqOptimized.scala:122)
[error] 	at scala.collection.LinearSeqOptimized.foldLeft$(LinearSeqOptimized.scala:118)
[error] 	at scala.collection.immutable.List.foldLeft(List.scala:86)
[error] 	at stryker4s.mutants.applymutants.MatchBuilder.buildNewSource(MatchBuilder.scala:17)
[error] 	at stryker4s.mutants.Mutator.buildMatches(Mutator.scala:44)
[error] 	at stryker4s.mutants.Mutator.$anonfun$mutate$1(Mutator.scala:21)
[error] 	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
[error] 	at scala.collection.immutable.List.foreach(List.scala:389)
[error] 	at scala.collection.TraversableLike.map(TraversableLike.scala:234)
[error] 	at scala.collection.TraversableLike.map$(TraversableLike.scala:227)
[error] 	at scala.collection.immutable.List.map(List.scala:295)
[error] 	at stryker4s.mutants.Mutator.mutate(Mutator.scala:18)
[error] 	at stryker4s.Stryker4s.run(Stryker4s.scala:16)
[error] 	at stryker4s.run.Stryker4sRunner$.delayedEndpoint$stryker4s$run$Stryker4sRunner$1(Stryker4sRunner.scala:23)
[error] 	at stryker4s.run.Stryker4sRunner$delayedInit$body.apply(Stryker4sRunner.scala:12)
[error] 	at scala.Function0.apply$mcV$sp(Function0.scala:34)
[error] 	at scala.Function0.apply$mcV$sp$(Function0.scala:34)
[error] 	at scala.runtime.AbstractFunction0.apply$mcV$sp(AbstractFunction0.scala:12)
[error] 	at scala.App.$anonfun$main$1$adapted(App.scala:76)
[error] 	at scala.collection.immutable.List.foreach(List.scala:389)
[error] 	at scala.App.main(App.scala:76)
[error] 	at scala.App.main$(App.scala:74)
[error] 	at stryker4s.run.Stryker4sRunner$.main(Stryker4sRunner.scala:12)
[error] 	at stryker4s.run.Stryker4sRunner.main(Stryker4sRunner.scala)
[error] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[error] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[error] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[error] 	at java.lang.reflect.Method.invoke(Method.java:498)
[error] 	at sbt.Run.invokeMain(Run.scala:93)
[error] 	at sbt.Run.run0(Run.scala:87)
[error] 	at sbt.Run.execute$1(Run.scala:65)
[error] 	at sbt.Run.$anonfun$run$4(Run.scala:77)
[error] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
[error] 	at sbt.util.InterfaceUtil$$anon$1.get(InterfaceUtil.scala:10)
[error] 	at sbt.TrapExit$App.run(TrapExit.scala:252)
[error] 	at java.lang.Thread.run(Thread.java:748)
[error] java.lang.RuntimeException: Nonzero exit code: 1
[error] 	at sbt.Run$.executeTrapExit(Run.scala:124)
[error] 	at sbt.Run.run(Run.scala:77)
[error] 	at sbt.Defaults$.$anonfun$bgRunTask$5(Defaults.scala:1185)
[error] 	at sbt.Defaults$.$anonfun$bgRunTask$5$adapted(Defaults.scala:1180)
[error] 	at sbt.internal.BackgroundThreadPool.$anonfun$run$1(DefaultBackgroundJobService.scala:366)
[error] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
[error] 	at scala.util.Try$.apply(Try.scala:209)
[error] 	at sbt.internal.BackgroundThreadPool$BackgroundRunnable.run(DefaultBackgroundJobService.scala:289)
[error] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error] 	at java.lang.Thread.run(Thread.java:748)
[error] (Compile / run) Nonzero exit code: 1
[error] Total time: 2 s, completed Oct 7, 2018 8:59:39 AM

```

**Comment by legopiraat** [comment]
@AlexITC Thanks for giving it another try, sadly this is a known bug.

Currently, when a pattern match exists on primitives types it will try and mutate them and will fail to compile because of that. I was looking through some of your code and I found the pattern match here: 
https://github.com/AlexITC/playsonify/blob/develop/playsonify/src/com/alexitc/playsonify/parsers/FieldOrderingParser.scala Line 58.

there is already an issue for this #48 sadly we did not find the time yet to fix it.

**Comment by hugo-vrijswijk** [comment]
It should be fixed in #69

**Comment by AlexITC** [comment]
Thanks for the info.

**Comment by hugo-vrijswijk** [comment]
I played around a bit and got your project working. I checked out the `literal-pattern-match-in-topstatement` branch to test it out before the PR is merged. I got some weird results with the first mutation always being applied, so I also added a clean command to Mill (which seriously increased the duration of the whole thing): 

```hocon
stryker4s {
  mutate: [
    "playsonify/src/**/*.scala"
  ]
  base-dir: "../playsonify"
  test-runner: {
    type: commandrunner
    command: "mill"
    args: "all clean playsonify.test"
  }
}
```

44 mutants are found and the whole thing took about 40 minutes on my machine. I've [uploaded the results here](https://gist.github.com/hugo-vrijswijk/44c5061518f762cea6563a4100fd64c2) so you don't have to do it again haha.

I'll close this issue now as it looks to be working (a Mill plugin to speed it up would be cool, of course, but I think that's a little further down the line, we're still working on our sbt plugin)
